src/components/DatePicker.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from "react";
+import {
+  View,
+  SafeAreaView,
+  StyleSheet,
+  StyleProp,
+  ViewStyle,
+} from "react-native";
+import DateTimePicker from "@react-native-community/datetimepicker";
+import dateFormat from "dateformat";
+import { withTheme } from "../core/theming";
+
+import Portal from "./Portal/Portal";
+import Button from "./Button";
+import TextField, { Props as TextFieldProps } from "./TextField";
+import Touchable from "./Touchable";
+import theme from "../styles/DefaultTheme";
+
+interface Props extends TextFieldProps {
+  style?: StyleProp<ViewStyle>;
+  theme: typeof theme;
+  // initialDate?: string;
+  // locale?: string;
+  // minuteInterval?: number;
+  // timeZoneOffsetInMinutes?: number;
+  // error?: boolean;
+  // type?: string;
+  date?: Date;
+  format?: string;
+  onDateChange?: (data?: any) => void;
+  disabled?: boolean;
+  mode?: "date" | "time" | "datetime";
+}
+
+const DatePicker: React.FC<Props> = ({
+  style,
+  theme: { colors },
+  date = new Date(),
+  onDateChange = () => {},
+  disabled = false,
+  mode = "date",
+  format,
+  ...props
+}) => {
+  const [pickerVisible, setPickerVisible] = React.useState(false);
+  const textField = React.useRef<typeof TextField | null>();
+
+  const formatDate = (): string => {
+    if (format) return dateFormat(date, format);
+
+    const months = [
```

Comment:
mind putting this outside of the component since its a constant? you can call it `const MONTHS =`